### PR TITLE
[ML] Instruction on how to run project from CLion

### PR DESCRIPTION
**Pull request by valeriy42**
This adds instructions on how to set up the CLion IDE to work with our project.

### CODE REVIEW

File: build-setup/clion/using_clion.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+and clicking the green play button. Moreover, we will create a configuration for running unit tests and use `Libraries`
+as a build dependency so we ensure that the project is up-to-date every time we run those tests.
+
+Let's create another **Run/Debug Configuration** for running uni tests for the `core` module. Again, go to **Run | Edit
+Configurations...** and click on the **+** symbol to create a new **Custom Build Application**:
+
+**Name:** Test core\
+**Target:** *Select the custom build target `make build` that we created before*
+
+**Executable:** *Navigate to `lib/core/unittest/` and select the `ml-test` binary*\
```
[the remark this answers is not in the document]
Comment (valeriy42) [reply build-setup/clion/using_clion.md]: fixed.

File: build-setup/clion/using_clion.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
+# Using CLion with `ml-cpp`
+
+CLion 2018.2 has added support for the compilation database project format. This tutorial shows how to create a
+compilation database for the `ml-cpp` project and integrate it with CLion.
+
+## Installing prerequisites
+
+>Note: this tutorial assumes that you have installed the correct version of GCC in your system environment and it can be
+>accessed by simply calling `g++`.
```
[the remark this answers is not in the document]
Comment (valeriy42) [reply build-setup/clion/using_clion.md]: Done. Thank you! I don't have hardware to test the setup for MacOS myself unfortunately.

**Comment by valeriy42** [comment]
Retest

**Comment by valeriy42** [comment]
I noticed, I still need to fix some desriptions before unittests will work.

**Comment by droberts195** [comment]
> I think very similar instructions would work on macOS too. I haven't had a change to try this

I still haven't had a chance to try this, but @edsavage has.  I suggest that @edsavage pushes the changes to the instructions that were required for macOS directly to the PR branch.  Then the PR can be merged and we should make CLion the recommended IDE for working on ml-cpp as it seems to be the best one we've tried.

**Pull request merged**